hal-core/src/mem/page/alloc.rs

```diff
@@ -0,0 +1,571 @@
+use super::{Alloc, AllocErr, PageRange, Size};
+use crate::{
+    mem::{Region, RegionKind},
+    Address, PAddr, VAddr,
+};
+use core::ptr;
+use mycelium_util::intrusive::{list, List};
+use mycelium_util::math::Log2;
+use mycelium_util::sync::{
+    atomic::{
+        AtomicUsize,
+        Ordering::{AcqRel, Acquire, Relaxed},
+    },
+    spin,
+};
+use mycelium_util::trace;
+
+#[derive(Debug)]
+pub struct BuddyAlloc<L = [spin::Mutex<List<Free>>; 32]> {
+    /// Minimum allocateable page size in bytes.
+    ///
+    /// Free blocks on free_lists[0] are one page of this size each. For each
+    /// index higher in the array of free lists, the blocks on that free list
+    /// are 2x as large.
+    pub min_size: usize,
```

Comment:
Why is this field `pub`? Feels like it probably shouldn't be given that mutating it would invalidate e.g. `min_size_log2`.

Reply:
It was intended for debug stats etc--- I thought it was okay because the allocator is immutable when it's in use (it locks the free lists internally). I should probably just make it an accessor though.